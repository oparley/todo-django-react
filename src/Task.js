import React, {Component} from 'react';

class Task extends Component{
    constructor(props){
        super(props);
        this.state = {
            name: props.name,
            completed: props.completed,
            id: props.id,
        };

        this.changeName = this.changeName.bind(this);
        this.markAsCompleted = this.markAsCompleted.bind(this);
        this.deleteTask = this.deleteTask.bind(this);
    }

    markAsCompleted(){
        this.setState({
            completed: !this.state.completed,
        });
    }

    changeName(event){
        this.setState({
            name: event.target.value
        });
    }

    deleteTask(event){
        let confirm = window.confirm("Are you sure?")
        if(confirm){
            this.props.callbackParent(this.state.id);
        }
    }

    render(){
        let text = <input
            type="text"
            className="form-control"
            onChange={this.changeName}
            value={this.state.name}
        />;

        if (this.state.completed) {
            text = <del className="form-control">{this.state.name}</del>;
        }

        return(
            <div className="input-group mb-3">
                {text}
                <div className="input-group-append">
                    <button onClick={this.markAsCompleted} className="btn btn-primary" type="button"><i className="far fa-check-square"></i></button>
                    <button onClick={this.deleteTask} className="btn btn-danger" type="button"><i className="far fa-trash-alt"></i></button>
                </div>
            </div>
        );
    }
}

export default Task;
