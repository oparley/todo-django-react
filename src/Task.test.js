import { MemoryRouter } from 'react-router'
import React from 'react';
import renderer from 'react-test-renderer';
import { shallow, mount, render } from 'enzyme';
import Task from './Task';

let task = {id: 1, name: 'test task'}
let taskPage = <MemoryRouter><Task task={task} listId='1'/></MemoryRouter>

describe('Task', () => {
    it('renders correctly', () => {
      const tree = renderer
        .create(taskPage)
        .toJSON();
      expect(tree).toMatchSnapshot();
    });

    it('shows task name in input', () => {
      const comp = mount(taskPage)
      expect(comp.find('a.button.input').text()).toEqual('test task')
    });

    it('shows task name in input', () => {
        task.completed = true
        const comp = mount(taskPage)
        expect(comp.find('del').exists())
    });

    it('shows task name in input', () => {
        task.completed = true
        const comp = mount(taskPage)
        expect(comp.find('del').text()).toEqual('test task')
    });
  })
