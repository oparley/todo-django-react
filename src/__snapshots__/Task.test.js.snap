// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Task renders correctly 1`] = `
<div
  className="field"
  tabIndex="0"
>
  <div
    className="field has-addons"
  >
    <div
      className="control is-expanded"
    >
      <a
        className="button input"
        href="lists/undefined/tasks/1"
        onClick={[Function]}
      >
        test task
      </a>
    </div>
    <div
      className="control"
    >
      <a
        className="button is-success"
        onClick={[Function]}
      >
        <i
          className="far fa-check-square"
        />
      </a>
    </div>
  </div>
</div>
`;
